humility-cmd/src/doppel.rs

```diff
@@ -38,12 +38,44 @@ use humility::reflect::{Load, Ptr, Value};
 use std::convert::TryInto;
 use zerocopy::{AsBytes, LittleEndian, U16, U64};
 
+// this struct exist to just help load the 32 bit task desc
 #[derive(Copy, Clone, Debug, Eq, PartialEq, Load)]
-pub struct TaskDesc {
+pub struct TaskDesc32 {
```

Comment:
Is it not possible to make use of `pub struct TaskDesc<T>` rather than having three different TaskDesc structs or would this create too much code churn?

Reply:
I was not sure if the `Load` trait will play nice with generic types, but I may be able to combine `TaskDesc32` and `TaskDesc64`.  I did not want `TaskDesc` to have have a generic though, otherwise every time I use the type, the associated `u32` or `u64` would have to be included.  By forcing everything to 64bit, the architecture is kind of abstracted out.

Reply:
Hm, I get the strangest of errors.
```
error[E0107]: missing generics for struct `TaskDescArch`
  --> humility-cmd/src/doppel.rs:43:12
   |
43 | pub struct TaskDescArch<T> {
   |            ^^^^^^^^^^^^ expected 1 generic argument
   |
note: struct defined here, with 1 generic parameter: `T`
  --> humility-cmd/src/doppel.rs:43:12
   |
43 | pub struct TaskDescArch<T> {
   |            ^^^^^^^^^^^^ -
help: add missing generic argument
   |
43 | pub struct TaskDescArch<T><T> {
```
This only occurs when deriving the `Load` trait.  It uses some macro magic that I need to look into.  I could also manually implement the trait, but I think the readability would be less.

Reply:
I played around with this some more, I think the `Load` trait will cause substantial problems with trying to combine all of these.  I think I will leave it as it.